mlkem/kem.c

```diff
@@ -120,32 +120,30 @@ static int mlk_check_sk(const uint8_t sk[MLKEM_INDCCA_SECRETKEYBYTES])
 }
 
 MLK_MUST_CHECK_RETURN_VALUE
-static int mlk_check_pct(uint8_t const pk[MLKEM_INDCCA_PUBLICKEYBYTES],
-                         uint8_t const sk[MLKEM_INDCCA_SECRETKEYBYTES])
+static int mlk_check_pct(const mlk_public_key *pk, const mlk_secret_key *sk)
 __contract__(
-  requires(memory_no_alias(pk, MLKEM_INDCCA_PUBLICKEYBYTES))
-  requires(memory_no_alias(sk, MLKEM_INDCCA_SECRETKEYBYTES)));
+  requires(memory_no_alias(pk, sizeof(mlk_public_key)))
+  requires(memory_no_alias(sk, sizeof(mlk_secret_key))));
 
 #if defined(MLK_KEYGEN_PCT)
 /* Specification:
  * Partially implements 'Pairwise Consistency Test' [FIPS 140-3 IG] and
  * [FIPS 203, Section 7.1, Pairwise Consistency]. */
 
 /* Reference: Not implemented in the reference implementation. */
-static int mlk_check_pct(uint8_t const pk[MLKEM_INDCCA_PUBLICKEYBYTES],
-                         uint8_t const sk[MLKEM_INDCCA_SECRETKEYBYTES])
+static int mlk_check_pct(const mlk_public_key *pk, const mlk_secret_key *sk)
```

Comment:
It is unclear if this will remain FIPS compliant -- we should keep the previous form of the PCT for now?